cmake-modules/AzureVcpkg.cmake

```diff
@@ -18,13 +18,50 @@ macro(az_vcpkg_integrate)
   endif()
 endmacro()
 
-macro(az_vcpkg_export targetName macroNamePart dllImportExportHeaderPath)
+macro(az_vcpkg_portfile_prep targetName fileName contentToRemove)
+  # with sdk/<lib>/vcpkg/<fileName>
+  file(READ "${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/${fileName}" fileContents)
+
+  # Windows -> Unix line endings
+  string(FIND fileContents "\r\n" crLfPos)
+  string(REPLACE "\r\n" "\n" fileContents ${fileContents})
+
+  # remove comment header
+  string(REPLACE "${contentToRemove}" "" fileContents ${fileContents})
+
+  # undo Windows -> Unix line endings (if applicable)
+  if (crLfPos GREATER -1)
+    string(REPLACE "\n" "\r\n" fileContents ${fileContents})
+  endif()
+  unset(crLfPos)
+
+  # output to an intermediate location
+  file (WRITE "${CMAKE_BINARY_DIR}/vcpkg_prep/${targetName}/${fileName}" ${fileContents})
+  unset(fileContents)
+
   # Produce the files to help with the VcPkg release.
   # Go to the /out/build/<cfg>/vcpkg directory, and copy (merge) "ports" folder to the vcpkg repo.
   # Then, update portfile.cmake's SHA512 from "1" to the actual hash (a good way to do it is to uninstall a package,
   # clean vcpkg/downloads, vcpkg/buildtrees, run "vcpkg install <pkg>", and get the SHA from the error message).
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/CONTROL" "${CMAKE_BINARY_DIR}/vcpkg/ports/${targetName}-cpp/CONTROL" @ONLY)
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/portfile.cmake" "${CMAKE_BINARY_DIR}/vcpkg/ports/${targetName}-cpp/portfile.cmake" @ONLY)
+  configure_file(
+    "${CMAKE_BINARY_DIR}/vcpkg_prep/${targetName}/${fileName}"
+    "${CMAKE_BINARY_DIR}/vcpkg/ports/${targetName}-cpp/${fileName}"
+    @ONLY
+  )
+endmacro()
+
+macro(az_vcpkg_export targetName macroNamePart dllImportExportHeaderPath)
+  az_vcpkg_portfile_prep(
+    "${targetName}"
+    "CONTROL"
+    "# Copyright (c) Microsoft Corporation. All rights reserved.\n# SPDX-License-Identifier: MIT\n#\n"
```

Comment:
Why is the last `\n` on a new line here (preceded by `#`) but not below for the portfile?

Reply:
See any CONTROL file - `#` is there as 3rd line. And it is not removable, in the terms that you cannot replace it with just empty string - VcPkg fails. What we can do is to get rid of that line entirely, but then it becomes too cluttered.

Reply:
See this file, for example - https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/core/azure-core/vcpkg/CONTROL

Reply:
Ah yes, just saw that. The checked-in files have that discrepancy. Thanks for clarifying why its there.